src/ExchangeSharp/API/Exchanges/Coinbase/ExchangeCoinbaseAPI.cs

```diff
@@ -10,1171 +10,816 @@ MIT LICENSE
 		{
-			if (marketSymbols == null || marketSymbols.Length == 0)
-			{
-				marketSymbols = (await GetMarketSymbolsAsync()).ToArray();
-			}
-			return await ConnectPublicWebSocketAsync(
-					"/",
-					async (_socket, msg) =>
-					{
-						JToken token = JToken.Parse(msg.ToStringFromUTF8());
-						if (token["type"].ToStringInvariant() == "error")
-						{ // {{ "type": "error", "message": "Failed to subscribe", "reason": "match is not a valid channel" }}
-							Logger.Info(
-												token["message"].ToStringInvariant()
-														+ ": "
-														+ token["reason"].ToStringInvariant()
-										);
-							return;
-						}
-						if (token["type"].ToStringInvariant() != "match")
-							return; //the ticker channel provides the trade information as well
-						if (token["time"] == null)
-							return;
-						ExchangeTrade trade = ParseTradeWebSocket(token);
-						string marketSymbol = token["product_id"].ToStringInvariant();
-						await callback(new KeyValuePair<string, ExchangeTrade>(marketSymbol, trade));
-					},
-					async (_socket) =>
+			// Legacy Orders may be retrieved using V2 (not implemented here - see GetTx in code below)
+			List<ExchangeOrderResult> orders = new List<ExchangeOrderResult>();
+			pagination = PaginationType.V3Cursor;
+			string startURL = "/orders/historical/fills";
+		
+			if (!string.IsNullOrEmpty(marketSymbol)) startURL += "?product_id=" + marketSymbol;
+			if (afterDate != null) startURL += marketSymbol == null ? "?" : "&" + "start_sequence_timestamp=" + ((DateTimeOffset)afterDate).ToUnixTimeSeconds();
+			JToken token = await MakeJsonRequestAsync<JToken>(startURL);
+			startURL +=  marketSymbol == null && afterDate == null ? "?" : "&" + "cursor=";
+			while(true)
+			{ 
+				foreach (JToken fill in token["fills"])
+				{ 
+					orders.Add(new ExchangeOrderResult() 
 					{
-						var subscribeRequest = new
-						{
-							type = "subscribe",
-							product_ids = marketSymbols,
-							channels = new object[]
-											{
-														new { name = "matches", product_ids = marketSymbols }
-									}
-						};
-						await _socket.SendMessageAsync(subscribeRequest);
-					}
-			);
+						MarketSymbol = fill[PRODUCTID].ToStringInvariant(),
+						TradeId = fill["trade_id"].ToStringInvariant(),
+						OrderId = fill["order_id"].ToStringInvariant(),
+						OrderDate = fill["trade_time"].ToDateTimeInvariant(),
+						IsBuy = fill[SIDE].ToStringInvariant() == BUY,
+						Amount = fill[SIZE].ConvertInvariant<decimal>(),
+						AmountFilled = fill[SIZE].ConvertInvariant<decimal>(),
+						Price = fill[PRICE].ConvertInvariant<decimal>(),
+						Fees = fill["commission"].ConvertInvariant<decimal>(),
+						AveragePrice = fill[PRICE].ConvertInvariant<decimal>()
+					});
+				}
+				if (string.IsNullOrEmpty(cursorNext)) break;
+				token = await MakeJsonRequestAsync<JToken>(startURL + cursorNext);
+			} 
+			pagination = PaginationType.None;
```

Comment:
[minor] This seems odd, shouldn't we store the previous value [here](https://github.com/DigitalRuby/ExchangeSharp/pull/818/files#diff-968c4bfa7ec1e7cbf9f9b6eac362a279e9269605ad5469344924a33fac413a2eR399) and then reset that value back to pagination after the v3cursor is processed instead of `.None`?